lib/spi.toit

```diff
@@ -120,6 +134,21 @@ interface Device extends serial.Device:
       --dc/int=0
       --command/int=0
       --address/int=0
+      --keep_cs_active/bool=false
+
+  /**
+  Reserves the bus for this device while executing the given $block.
+
+  Starts by acquiring the bus. Once that's succeeded, executes the $block. Finally, releases
+    the bus before returning.
+
+  Reserving the bus can be useful in two contexts:
+  1. The CS pin is controlled by the user. Since the hardware only supports a limited number of
+    automatic CS pins, it might be necessary to set some CS pins by hand. This should be done
+    after the bus has been reserved.
+  2. When using the `--keep_cs_active` flag of the $transfer function, the bus must be reserved.
+  */
+  reserve_bus [block]
```

Comment:
Alternative names that makes it slightly more obvious that this takes a block argument:

reserve_bus_during
reserved_do
with_reserved_bus

Reply:
Went with `with_reserved_bus`.